File: app/src/main/java/com/axborn/androidparty/features/loading/LoadingActivity.java
package com.axborn.androidparty.features.loading;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import com.axborn.androidparty.R;

public class LoadingActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_loading);

        String token = getIntent().getStringExtra("TOKEN");
        System.out.println(token);
    }
}

File: app/src/main/java/levkovskiy/dev/tesonettest/ui/loading/LoadingActivity.kt
package levkovskiy.dev.tesonettest.ui.loading

import android.content.Intent
import android.os.Bundle
import io.reactivex.Observable
import io.reactivex.android.schedulers.AndroidSchedulers
import levkovskiy.dev.tesonettest.R
import levkovskiy.dev.tesonettest.ui.base.BaseActivity
import levkovskiy.dev.tesonettest.ui.main.MainActivity
import java.util.concurrent.TimeUnit

class LoadingActivity : BaseActivity() {
  override fun getLayoutRes() = R.layout.activity_loading

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)

    Observable.timer(5000, TimeUnit.MILLISECONDS)
        .observeOn(AndroidSchedulers.mainThread())
        .subscribe {
          startNextScreen()
          finish()
        }
  }

  private fun startNextScreen() {
    val intent = Intent(this, MainActivity::class.java)
    startActivity(intent)
  }
}
